Framework/DataObjects/src/BasePeak.cpp

```diff
@@ -0,0 +1,477 @@
+/** Return the run number this peak was measured at. */
+int BasePeak::getRunNumber() const { return m_runNumber; }
+
+/** Set the run number that measured this peak
+ * @param m_runNumber :: the run number   */
+void BasePeak::setRunNumber(int m_runNumber) {
+  this->m_runNumber = m_runNumber;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Return the monitor count stored in this peak. */
+double BasePeak::getMonitorCount() const { return m_monitorCount; }
+
+/** Set the monitor count for this peak
+ * @param m_monitorCount :: the monitor count */
+void BasePeak::setMonitorCount(double m_monitorCount) {
+  this->m_monitorCount = m_monitorCount;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Get the final neutron energy in meV */
+double BasePeak::getFinalEnergy() const { return m_finalEnergy; }
+
+/** Get the initial (incident) neutron energy in meV */
+double BasePeak::getInitialEnergy() const { return m_initialEnergy; }
+
+/** Get the difference between the initial and final neutron energy in meV */
+double BasePeak::getEnergyTransfer() const {
+  return getInitialEnergy() - getFinalEnergy();
+}
+
+//----------------------------------------------------------------------------------------------
+/** Get the H index of the peak */
+double BasePeak::getH() const { return m_H; }
+
+/** Get the K index of the peak */
+double BasePeak::getK() const { return m_K; }
+
+/** Get the L index of the peak */
+double BasePeak::getL() const { return m_L; }
+
+/** Return the HKL vector */
+Mantid::Kernel::V3D BasePeak::getHKL() const { return V3D(m_H, m_K, m_L); }
+
+/** Return True if the peak has been indexed */
+bool BasePeak::isIndexed() const {
+  if (m_H == 0. && m_K == 0. && m_L == 0.)
+    return false;
+  return true;
+}
+
+/** Return the int HKL vector */
+Mantid::Kernel::V3D BasePeak::getIntHKL() const { return m_intHKL; }
+
+/** Return the int MNP vector */
+V3D BasePeak::getIntMNP() const { return m_intMNP; }
+
+//----------------------------------------------------------------------------------------------
+/** Set the H index of this peak
+ * @param m_H :: index to set   */
+void BasePeak::setH(double m_H) { this->m_H = m_H; }
+
+/** Set the K index of this peak
+ * @param m_K :: index to set   */
+void BasePeak::setK(double m_K) { this->m_K = m_K; }
+
+/** Set the L index of this peak
+ * @param m_L :: index to set   */
+void BasePeak::setL(double m_L) { this->m_L = m_L; }
+
+/** Set the BankName of this peak
+ * @param m_bankName :: index to set   */
+void BasePeak::setBankName(std::string m_bankName) {
+  this->m_bankName = std::move(m_bankName);
+}
+
+/** Set all three H,K,L indices of the peak */
+void BasePeak::setHKL(double H, double K, double L) {
+  m_H = H;
+  m_K = K;
+  m_L = L;
+}
+
+/** Set all HKL
+ *
+ * @param HKL :: vector with x,y,z -> h,k,l
+ */
+void BasePeak::setHKL(const Mantid::Kernel::V3D &HKL) {
```

Comment:
This should be something like
```C++
this->setHKL(HKL.X(), HKL.Y(), HKL.Z());
```
so when somebody decides we need argument checking it happens in one place.